README.md

```diff
@@ -29,6 +30,206 @@ runApp(...
+
+The tool assumes that, with proper memory management,
+an object's disposal and garbage collection should happen sequentially,
+close to each other.
+
+By monitoring disposal and GC events, the tool detects different types of leaks:
+
+**Not disposed, but GCed (not-disposed)**: a disposable object was GCed,
+without being disposed first. This means that the object's disposable content
+is using memory after the object is no longer needed.
+To fix the leak, invoke `dispose()` to free up the memory.
+
+**Disposed, but not GCed (not-GCed)**: an object was disposed,
+but not GCed after certain number of GC events. This means that
+a reference to the object is preventing it from being
+garbage collected after it's no longer needed.
+To fix the leak, after disposal assign all reachable references
+of the object to null:
+
+```
+myField.dispose();
+myField = null;
+```
+
+**Disposed and GCed late (GCed-late)**: an object was disposed and then GCed,
+but GC happened later than expected. This means the retaining path was
+holding the object in memory for some period, but then disappeared.
+
+**Disposed, but not GCed, without path (not-GCed-without-path)**: an object
+was disposed and not GCed when expected, but retaining path is not detected,
+that means that the object will be most likely GCed in the next GC cycle,
+and the leak will convert to **GCed-late** leak.
+
+### Culprits and victims
+
+If you have a set of not-GCed objects, some of them (victims)
+might not be GC-ed because they are held by others (culprits).
+Normally, to fix the leaks, you need to only fix the culprits.
+
+**Victim**: a leaked object, for which the tool could find another
+leaked object that, if fixed, would also fix the first leak.
+
+**Culprit**: a leaked object that is not detected to be the victim
+of another object.
+
+The tool detects which leaked objects are culprits, so you know where to focus.
+
+For example, out of four not-GCed leaks on the following diagram,
+only one is the culprit, because, when the object is fixed
+and GCed, the victims it referenced will be also GCed:
+
+```mermaid
+
+   flowchart TD;
+      l1[leak1\nculprit]
+      l2[leak2\nvictim]
+      l3[leak3\nvictim]
+      l4[leak4\nvictim]
+      l1-->l2;
+      l1-->l3;
+      l2-->l4;
+      l3-->l4;
+```
+
+## Limitations
+
+### By tracked classes
+
+The leak traker will catch leaks only for instrumented onjects (See [concepts](#leak-tracking-concepts) for details).
+
+However, the good news are:
+
+1. Most of disposable Flutter Framework classes are already instrumented. So, Widget related
+leaks in your Flutter application will be cought without additional effort.
+
+2. If a leak involves at least one instrumented object, the leak will be caught and all
+other objects, even non-instrumented, will stop leaking as well.
+
+See [the instrumentation guidance](#instrument-your-code).
+
```

Comment:
It might be a good idea to add a section mentioning how to report a bug if they notice that a framework entity has a leak

Reply:
Thanks. Updated the sentence to show it is most likely in-app problem.